internal/util/dirent.go

```diff
@@ -48,6 +52,10 @@ func CopyLink(source string, dest string) error {
 		dest = filepath.Join(dest, filepath.Base(source))
 	}
 
+	if source == dest {
```

Comment:
These need to be normalized before they are compared.
I would just do ShouldAbs() to normalize them (filepath.Abs() calls filepath.Clean()).
Note that we can abs here because we are just about to do the operation on them
(whereas FileOps and FileSpecs have to hang around for a while until they can be evaluated).